python/hsfs/core/feature_group_engine.py

```diff
@@ -99,6 +123,10 @@ def insert(
                 "Online storage is not enabled for this feature group."
             )
 
+        if not feature_group._id:
+            # this means FG doesn't exist and should create the new one
+            self.save(feature_group, feature_dataframe, write_options)
```

Comment:
I think here we are inserting the data twice. The save is going to trigger the `save_dataframe` method of the engine and and then the insert is going to trigger the same.